drivers/misc/bh_fwtable/bh_fwtable.c

```diff
@@ -216,49 +201,26 @@ static int tt_bh_fwtable_load(const struct device *dev, enum bh_fwtable_e table)
 	return 0;
 }
 
-static int tt_bh_fwtable_load_tables(const struct device *dev)
+static int tt_bh_fwtable_init(const struct device *dev)
 {
-	struct bh_fwtable_data *data = dev->data;
-
-	if (data->initialized) {
-		return 0; /* Already loaded */
-	}
-
 	/* load firmware tables from flash */
-	int result;
-
 	if (IS_ENABLED(CONFIG_TT_SMC_RECOVERY)) {
```

Comment:
The bits that need to be done just above this are

* check if the flash device is ready
* find offsets / addresses for "files"

Those addresses could be passed-in to the load function calls below.

W.r.t. boot fs, it would be kind of nice to not need a cache for it 🤷🏼‍♂️

With the rate that we access it, it seems non-ideal to allocate memory for it statically.